src/core/server/deprecations/README.mdx

```diff
@@ -0,0 +1,194 @@
+
+## Where does the deprecations show for the users?
+The Upgrade Assistant (UA) in kibana is consuming the core deprecations service in order to surface plugin deprecations. Each deprecation should have manual steps that we display in the UI to fix the deprecation an API to automatically resolve it.
+
+To check your plugin deprecations go to the UA interface and click on `view deprecations` under the kibana section.
+
+## How do I use this service for my deprecated plugin configurations?
+The deprecations service automatically hooks deprecated configs with the deprecations service. 
+
+All the config deprecation functions (`unused`, `unusedFromRoot`, `rename`, `renameFromRoot`) now accept an optional parameter to customize the deprecation details.
+
+### Example
+
+```ts
+export const config: PluginConfigDescriptor<ConfigType> = {
+  schema: configSchema,
+  deprecations: ({ renameFromRoot }) => [
+    renameFromRoot('ui_metric.debug', 'usageCollection.uiCounters.debug', {
+      documentationUrl: 'elastic.co/some-url',
+    }),
+  ],
+};
+```
+
+If the above deprecated kibana config `ui_metric.debug` is being used by the cluster.
+The service will show the following details about the deprecation:
+
+```ts
+{
+  "deprecationsInfo":[{
+    "level":"critical",
+    "message": "\"ui_metric.debug\" is deprecated and has been replaced by \"usageCollection.uiCounters.debug\"",
+    "documentationUrl": 'elastic.co/some-url',
+    "correctiveActions":{
+      "manualSteps": [
+        `Replace "ui_metric.debug" with "usageCollection.uiCounters.debug" in the Kibana config file, CLI flag, or environment variable (in Docker only).`,
+      },
+    "domainId":"usageCollection",
+  }],
+}
+```
+
+#### Custom deprecations for plugin configs
+Custom config deprecation handling allows specifying the deprecation details via the `addDeprecation`.
+
+##### Example:
+
+```ts
+export const config: PluginConfigDescriptor<ConfigSchema> = {
+  exposeToBrowser: {
+    defaultAppId: true,
+  },
+  schema: configSchema,
+  deprecations: () => [
+    (
+      completeConfig: Record<string, any>,
+      rootPath: string,
+      addDeprecation: AddConfigDeprecation
+    ) => {
+      if (
+        get(completeConfig, 'kibana.defaultAppId') === undefined &&
+        get(completeConfig, 'kibana_legacy.defaultAppId') === undefined
+      ) {
+        return completeConfig;
+      }
+      addDeprecation({
+        message: `kibana.defaultAppId is deprecated and will be removed in 8.0. Please use the \`defaultRoute\` advanced setting instead`,
+        correctiveActions: {
+          manualSteps: [
+            'Go to Stack Management > Advanced Settings',
+            'Update the "defaultRoute" setting under the General section',
+            'Remove "kibana.defaultAppId" from the kibana.yml config file',
+          ],
+        },
+      });
+      return completeConfig;
+    },
+  ],
+};
+```
+
+## How do I use this service for my deprecated plugin features?
```

Comment:
I think this title is pretty vague. Maybe something like "How do I add deprecations for non-config deprecations?"

It may also help to include a list of examples of when to use this API (eg. deprecated SO types, etc.) and use cases that should not use this API (eg. a deprecated HTTP endpoint being used by an external system can't be 'detected' by this mechanism and should use a deprecation log instead; though I don't think we have this yet)

Reply:
You can find the current list of breaking changes in Kibana [here](https://github.com/elastic/kibana/issues?q=is%3Aopen+is%3Aissue+label%3A%22Breaking+Change%22+label%3A%22Feature%3AUpgrade+Assistant%22). 

Some examples: deprecated roles (https://github.com/elastic/kibana/issues/81674, https://github.com/elastic/kibana/issues/92485), removal of coordinate map and regional map visualizations (https://github.com/elastic/kibana/issues/81703), removal of expression-based scripted fields (https://github.com/elastic/kibana/issues/77669).